### ISSUE TITLE
SC2031 is reported even if the subshell is in a different code path

### ISSUE THREAD

**Issue by ale5000-git** [koalaman/shellcheck#3068]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2031
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env sh
# shellcheck enable=all

my_func()
{
  if test "${1-}" = '-' && test "${#}" -eq 1; then
    (
      # shellcheck disable=SC2030 # Intended: Modification of LC_ALL is local (to subshell)
      LC_ALL='C'
      export LC_ALL

      some_code
    )
  else
    _old_lcall="${LC_ALL-}"
    LC_ALL='C'
    export LC_ALL

    some_other_code

    if test -n "${_old_lcall?}"; then LC_ALL="${_old_lcall:?}"; else unset LC_ALL; fi
  fi
}

my_func '-'
```

#### Here's what shellcheck currently says:
```
[Line 15:](javascript:setPosition(15, 17))
    _old_lcall="${LC_ALL-}"
                ^-- [SC2031](https://www.shellcheck.net/wiki/SC2031) (info): LC_ALL was modified in a subshell. That change might be lost.
```

#### Here's what I wanted or expected to see:

Nothing.
Since the subshell is in a different code path it cannot affect the code al line 15 so it shouldn't report it.
